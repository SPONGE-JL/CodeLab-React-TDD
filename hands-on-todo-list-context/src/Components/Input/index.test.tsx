import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { Input } from './index';

describe('<Input />', () => {
  it('reders components correctly.', () => {
    const { container } = render(<Input value="This is an input tag." />);

    const input = screen.getByDisplayValue('This is an input tag.');
    expect(input).toBeInTheDocument();

    expect(input).toMatchSnapshot();
  });

  it('renders with a specific placeholder.', () => {
    const { container } = render(<Input placeholder="Please something to insert" />);

    const input = screen.getByPlaceholderText('Please something to insert') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'Yeah-!' } });
    expect(input.value).toBe('Yeah-!');
  });
});
