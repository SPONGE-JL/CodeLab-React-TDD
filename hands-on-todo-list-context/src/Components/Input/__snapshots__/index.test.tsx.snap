// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> reders components correctly. 1`] = `
.c0 {
  font-size: 16px;
  text-align: center;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
  padding: 10px;
  margin-right: 10px;
}

<input
  class="c0"
  value="This is an input tag."
/>
`;
